push/src/gateway.cc

```diff
@@ -164,4 +164,10 @@ void Gateway::CleanupStalePointers(
       std::end(collectables));
 }
 
+int Gateway::AddHttpHeader(const std::string& header)
+{
+  std::lock_guard<std::mutex> lock{mutex_};
```

Comment:
I think we don't need to lock here because this mutex protects the collectables.